src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+    while (iterator.hasNext) {
+      val ctx = iterator.next()
+      val phaseSetId: Int = PhasingUtils.getPhasingSetId(ctx)
+      if (phaseSetId > 0) {
+        if (ctx.getContig != curContig) {
+          detector.addAll(phaseBlocks.values.toIterator)
+          phaseBlocks.clear()
+          curContig = ctx.getContig
+        }
+        val phaseBlock = phaseBlocks.get(phaseSetId) match {
+          case Some(block) =>
+            require(block.getContig == ctx.getContig)
+            require(block.getStart <= ctx.getStart)
+            new PhaseBlock(contig=ctx.getContig, start=block.getStart, end=Math.max(block.getEnd, ctx.getEnd)) // extend the block
+          case None        =>
+            new PhaseBlock(contig=ctx.getContig, start=ctx.getStart, end=ctx.getEnd)
+        }
+        phaseBlocks.put(phaseSetId, phaseBlock)
+      }
+      progress.record(ctx.getContig, ctx.getStart)
+    }
+
+    // add any remaining
+    detector.addAll(phaseBlocks.values.toIterator)
+    phaseBlocks.clear()
+
+    detector
+  }
+}
+
+case class PhaseBlock private (contig: String, start: Int, end: Int) extends Locatable {
+  def getContig: String = contig
+  def getStart: Int = start
+  def getEnd: Int = end
+  def length: Int = end - start + 1
+}
+
+object AssessPhasingMetric {
+  val MetricExtension = ".assess_phasing_metrics.txt"
+}
+
+/** Some counts about phasing
+  *
+  * @param num_called the number of variants called.
+  * @param num_phased the number of variants called with phase.
+  * @param num_truth the number of variants with known truth genotypes.
+  * @param num_truth_phased the number of variants with known truth genotypes with phase.
+  * @param num_called_with_truth_phased the number of variants called that had a known phased genotype.
+  * @param num_phased_with_truth_phased the number of variants called with phase that had a known phased genotype.
+  * @param num_truth_phased_in_called_block the number of known phased variants that were in a called phased block.
+  * @param num_both_phased_in_called_block the number of called phase variants that had a known phased genotype in a called phased block.  TODO: is this the same as `num_phased_with_truth_phased`?
+  * @param num_short_switch_errors the number of short switch errors (isolated switch errors).
+  * @param num_long_switch_errors the number of long switch errors (# of runs of consecutive switch errors).
+  * @param num_long_switch_sites the number of sites that could be long switch errors.
+  * @param num_illumina_point_switch_errors the number of point switch errors (defined in http://dx.doi.org/10.1038%2Fng.3119).
+  * @param num_illumina_long_switch_errors the number of long switch errors (defined in http://dx.doi.org/10.1038%2Fng.3119).
+  * @param num_illumina_long_switch_sites the number of sites that could be long switch errors (defined in http://dx.doi.org/10.1038%2Fng.3119).
+  * @param frac_phased the fraction of called variants with phase.
```

Comment:
I wanted to avoid having a ton of metrics (I failed).  But it would be
```
short_accuracy = 1 - (num_short_switch_errors / num_switch_sites)
long_accuracy = 1 - (num_long_switch_errors / num_switch_sites)
illumina_point_accuracy = 1 - (num_illumina_point_switch_errors / num_illumina_switch_sites )
illumina_long_accuracy = 1 - (num_illumina_long_switch_errors / num_illumina_switch_sites )
```